tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -452,3 +453,24 @@ def simple_broadcast(inp: Field[[IDim], float64]) -> Field[[IDim, JDim], float64
     simple_broadcast(a, out=out, offset_provider={"Joff": JDim})
 
     assert np.allclose(a.array()[:, np.newaxis], out)
+
+
+def test_domain_as_symbol():
+    size = 10
+    a = np_as_located_field(IDim)(np.ones((size)))
+    b = np_as_located_field(IDim)(np.zeros((size)))
+
+    @field_operator
+    def copy(inp: Field[[IDim], float64]) -> Field[[IDim], float64]:
+        return inp
+
+    @program(backend="roundtrip")
+    def copy_program(dom: Domain[IDim], inp: Field[[IDim], float64], out: Field[[IDim], float64]):
+        copy(inp, out=out[dom])
+
+    domain = {IDim: range(1, 9)}
```

Comment:
This is what I already used in embedded execution to represent a domain. Open to change that (now or later).

Reply:
Seems fine for now. Only reason to change that I see would be to simplify type deduction?